storagetransfer/transfer_from_aws.go

```diff
@@ -0,0 +1,95 @@
+package storagetransfer
+
+// [START storagetransfer_transfer_from_aws]
+import (
+	"context"
+	"fmt"
+	"google.golang.org/genproto/googleapis/type/date"
+	"google.golang.org/genproto/googleapis/type/timeofday"
+	"io"
+	"time"
+
+	storagetransfer "cloud.google.com/go/storagetransfer/apiv1"
+	storagetransferpb "google.golang.org/genproto/googleapis/storagetransfer/v1"
+)
+
+func transfer_from_aws(w io.Writer, projectID string, jobDescription string, awsSourceBucket string, gcsSinkBucket string, startTime time.Time, awsAccessKeyID string, awsSecretKey string) (*storagetransferpb.TransferJob, error) {
+	// Your Google Cloud Project ID
+	// projectID := "my-project-id"
+
+	// A description of this job
+	// jobDescription = "Transfers objects from an AWS bucket to a GCS bucket"
+
+	// The name of the Aws bucket to transfer objects from
+	// awsSourceBucket := "my-source-bucket"
+
+	// The name of the GCS bucket to transfer objects to
+	// gcsSinkBucket := "my-sink-bucket"
+
+	// The time to start the transfer
+	// startTime := time.Now()
```

Comment:
For jobDescription, startTime and the creds, it probably makes more sense to declare them inside the function rather than passing them in via arguments. Doesn't look necessary for testing in this case. Same comment applies to other samples.

https://googlecloudplatform.github.io/samples-style-guide/#minimal-arguments

Reply:
Thanks for pointing me there!

As an aside, this style guide also says to avoid return types whenever possible, which is what Java does, every GCS java sample is a `void`. 

I have return types here because (I believe) that's what the GCS Go samples do. Is there some reason for that, or is it just historical?

Reply:
Good question! Returning an `error` is just idiomatic error handling for Go, so we decided to make that an exception to the rule (after a lot of debate). The canonical snippet samples for Go have an error returned from all methods.

However, you're right that returning values other than errors is discouraged-- so, probably better to avoid it unless it's absolutely necessary for testing.